client/source/js/modules/ui/gridster/resizable-chart-panel-directive.js

```diff
@@ -0,0 +1,102 @@
+define(['./module', 'underscore'], function (module, _) {
+  /**
+   * Adds a gridster directive to the element.
+   *
+   * The grid setup is a 16 column setup with 5x4 (5 rows 4 cols) per item and
+   * is currently not customizable. The directive applies appropriate chart resizing
+   * specific to stackedAreaCharts and lineAreaScatterCharts.
+   *
+   * The current setup should be extended to be more customizable as use cases expand.
+   */
+  return module
+      .directive('resizableChartPanel', function () {
+        /**
+         * We use the chart id to track by id in ng-repeat to avoid redrawing of all the charts when the chart array
+         * is updated when resizing the charts.
+         * */
+        var generateChartId = function () {
+          return _.uniqueId("chart_");
+        };
+
+        var resizeChart = function (chart, newWidth, newHeight) {
+          var widthOffset = 20;
+          var heightOffset = chart.type == 'stackedAreaChart' ? 210 : 100;
```

Comment:
@dragansah I believe going for a fixed height is bad choice here. There can be many more populations (which result in more labels). They height should be calculated depending on the amount or in best case dynamically like the normalise-height directive does.